ls1intum/Artemis
# [Feature] - Import exam quizzes as new quizzes

### Checklist

### Motivation and Context


It is already possible to import quizzes from courses. It would be very convenient to be able to import a quiz from an already existing exam. 

### Description

- It's now possible to import exam quiz questions from other exams where the user has instructor rights. If the user is an admin, he/she can import from every exam in the database.
- Added client side tests to test client side changes
- Added server side tests to test server additions

### Steps for Testing

Make sure that you test it with an Admin account, where you should be able to import every exam there is (pretty much to choose from in the testing server)
-> **Make also sure to test with an non-admin account that you can only access exams in which course you are also instructor**

1. Log in to Artemis
2. Create a new Quiz exercise
3. While creating, click on "Add existing question"
4. Choose the button "From an exam"
5. Choose an exam which already has an quiz question
6. Mark the question you want to import in the right hand side and click "Add Selected Questions"
7. The same question(s) as in your other exam should be added to your quiz
8. Also check that Importing from an course and from a file still works
9. Check all buttons and checkmarks (Like the ones for filtering) still work correctly
10. (You can also for example choose an exam without quizzes and in order to make sure that this is handled correctly as well)
11. Check that for the import from courses, no courses are shown which do not contain a quiz exercise

### Screenshots


<img width="1419" alt="Screenshot 2021-02-15 at 14 37 33" src="https://user-images.githubusercontent.com/44401560/107953778-df2fd500-6f9b-11eb-9127-3345fa6814c0.png">
